tribits/core/package_arch/TribitsGlobalMacros.cmake

```diff
@@ -254,12 +254,9 @@ MACRO(TRIBITS_DEFINE_GLOBAL_OPTIONS_AND_DEFINE_EXTRA_REPOS)
     "Enable the C++ compiler and related code"
     ${${PROJECT_NAME}_ENABLE_CXX_DEFAULT} )
 
-  IF ("${${PROJECT_NAME}_ENABLE_CXX11_DEFAULT}" STREQUAL "")
-    SET(${PROJECT_NAME}_ENABLE_CXX11_DEFAULT OFF)
-  ENDIF()
-  ADVANCED_OPTION(${PROJECT_NAME}_ENABLE_CXX11
-    "Enable the C++11 compiler options and related code (see ${PROJECT_NAME}_CXX11_FLAGS)"
-    ${${PROJECT_NAME}_ENABLE_CXX11_DEFAULT} )
+  # Hard-code a variable with the same name as a now-removed option that is always enabled.
+  # This can be removed after clients have been updated.
+  SET(${PROJECT_NAME}_ENABLE_CXX11 ON)
```

Comment:
It would be nice to get rid of this at some point since this is going to be confusing for projects that support/require C++ standards newer than C++11, like C++14 (see trilinos/Trilinos#6260).  But since downstream projects may still have logic that looks for this, I guess it is a good idea for now to keep defining this.

It would be nice if CMake had a way to deprecate a CMake variable so that if the variable got read by client code then it would generate a deprecated warning.  Otherwise, I don't know how to get a client to stop using a CMake variable of a given name :-(